tasks/build-binary-obs/lib/obs_binary_builder.rb

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env ruby
+
+require 'tempfile'
+require 'erb'
+require 'uri'
+
+class ObsBinaryBuilder
+  attr_accessor :binary, :version
+
+  TEMPLATE_PATH = File.join(File.dirname(__FILE__), "../templates")
+
+  def initialize(binary, version)
+    @binary = binary
+    @version = version
+  end
+
+  def build
+    package_string = "#{@binary}-#{@version}"
+
+    puts 'Create the package on OBS using "osc"'
+    create_obs_package(package_string)
+
+    puts 'Checkout the package with osc'
+    checkout_obs_package(package_string)
+
+    puts 'Removing obsolete old packages'
+    remove_obsolete_file_from_obs_project(package_string)
+
+    puts 'Change working directory'
+    Dir.chdir("#{obs_project}/#{package_string}")
+
+    puts 'Download the source and put them in the package dir'
+    fetch_sources
+
+    puts 'Render the spec template and put it in the package dir'
+    render_spec_template
+
+    puts 'Commit the changes on OBS'
+    commit_obs_package
+
+    puts 'Done!'
+  end
+
+  def source_url
+    case @binary
+    when 'bundler'
+      "http://rubygems.org/gems/#{@binary}-#{@version}.gem"
+    when 'ruby'
+      "https://cache.ruby-lang.org/pub/ruby/#{minor_version}/ruby-#{@version}.tar.gz"
+    end
+  end
+
+  private
+
+  def fetch_sources
+    run_command("wget #{source_url}")
+  end
+
+  def create_obs_package(package_name)
+    package_meta_template = <<EOF
```

Comment:
Maybe we can put this to a file in TEMPLATE_PATH if it grows, to improve readability. For now it's ok here.